### PR TITLE
feat: add `only-validate` param to validate quests on prs

### PR DESCRIPTION
**Pull request by darmalovan**
Adds the ability to validate quests on PRs
[Engine PR](https://github.com/trywilco/wilco-engine/pull/1870)

image

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -12387,41 +12387,63 @@ const zipQuest = async () => {
   return zipFile;
 }
 
-const uploadQuest = async (questId, zipFile) => {
-  console.log(`Uploading ${questId}`);
+const uploadQuest = async (questId, zipFile, onlyValidate = false) => {
```
Comment (esapir) [inline dist/index.js]: why dist is here?

File: index.js [hunk 2]
```diff
@@ -19,41 +19,63 @@ const zipQuest = async () => {
   return zipFile;
 }
 
-const uploadQuest = async (questId, zipFile) => {
-  console.log(`Uploading ${questId}`);
+const uploadQuest = async (questId, zipFile, onlyValidate = false) => {
```
Comment (esapir) [inline index.js]: The function is uploadQuest which should upload. It makes less sense to have a `onlyValidate` param. And the same in the Engine. I think we need a `validate` route in the Engine and a `validate` function here

File: dist/index.js (hunk 1 above)
Comment (darmalovan) [reply dist/index.js]: this done automatically by github

File: index.js [hunk 3]
```diff
@@ -19,41 +19,73 @@ const zipQuest = async () => {
...
+    console.log(resJson.error?.message);
+    throw new Error(resJson.message);
+  } else {
+    console.log("Draft was created successfully")
+    console.log(await res.json());
+  }
+}
+
+const validateQuest = async (questId, zipFile) => {
+  console.log(`Uploading ${questId}`);
```
Comment (esapir) [inline index.js]: validating

File: dist/index.js (hunk 1 above)
Comment (esapir) [reply dist/index.js]: Can you just set dist in .ignore?

**Review by esapir — APPROVED** [review]

File: dist/index.js (hunk 1 above)
Comment (darmalovan) [reply dist/index.js]: Don't think it's gonna help, It is github's commit

**Pull request merged**